### PR TITLE
[Source-mssql] : Remove options for data_to_sync & snapshot_isolation

### PR DESCRIPTION
**Pull request by akashkulk**
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2814 and https://github.com/airbytehq/airbyte/issues/33229

Main changes here are to remove options for data_to_sync & snapshot_isolation options from the source-mysql CDC config. 

1. **Data To Sync** : Previously, users were given an option to sync "Existing & New" data (snapshot + incremental changes) or "New Changes Only". With the snapshotting logic added in https://github.com/airbytehq/airbyte/pull/33018, we always end up performing a snapshot. IMO we should take an opinionated stance here to always snapshot if needed. This will ensure feature parity between other certified connectors. 

Previously, the `[ snapshot.mode](https://debezium.io/documentation/reference/stable/connectors/sqlserver.html#sqlserver-property-snapshot-mode)` property was set to : 
```
snapshot.mode = initial_only // For initial snapshot
snapshot.mode = initial OR schema_only // For incremental snapshots, based on configs
```

The main change here is to set 
```
snapshot.mode = initial_only // For initial snapshot
snapshot.mode = initial // For incremental snapshots
```

2.  **Snapshot isolation** : Previously, users were given an option to set snapshot isolation. This would only be relevant if the user was set up in "Existing And New" mode. I propose that the [snapshot.isolation](https://debezium.io/documentation/reference/stable/connectors/sqlserver.html#sqlserver-property-snapshot-isolation-mode) property be set to `read_committed` by default. Airbyte's use case does not require the `snapshot` isolation level, which may lock tables while reading, but ensures full consistency. Airbyte ensure "at-least-once" data deliver and not "exactly once" data delivery. In addition, with the work to build our own snapshotting mechanism, this setting will soon be obsolete. 

**User impact**
1. Changing the snapshot isolation will not have any effect on user syncs. 
2. Changing the default mode to snapshot when the WAL has rotated may trigger refresh syncs for users, but we have no active users in cloud who are using this sync mode

### CODE REVIEW

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java [hunk 1]
```diff
@@ -165,10 +80,12 @@ static Properties getDebeziumProperties(final JdbcDatabase database, final Confi
     if (isSnapshot) {
       props.setProperty("snapshot.mode", "initial_only");
     } else {
-      props.setProperty("snapshot.mode", getDataToSyncConfig(config).getDebeziumSnapshotMode());
+      // If not in snapshot mode, initial_only will make sure that a snapshot is taken if the oplog is
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java:83]: nit: is it Oplog in SQL Server?

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java
Comment (rodireich) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java:1]: Should this be communicated to customers before releasing?

**Review by rodireich — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java (hunk 1 above)
Comment (nguyenaiden) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java:83]: iirc it's `transaction_log` in MSSQL

**Review by nguyenaiden — APPROVED** [review]
+1. We should reach out to that one customer and let them know.

**Review by stephane-airbyte — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java (hunk 1 above)
Comment (akashkulk) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java]: Done

**Comment by postamar** [comment]
Should you also remove some methods from the MsSqlTestDatabase implementation in the test fixtures? IIRC there's something about snapshot isolation in there.

**Pull request merged**